From: "Jason R. Coombs" <jaraco@jaraco.com>
Date: Tue, 11 Jan 2022 18:04:05 -0400
Subject: Fix install_scripts() if bdist_wininst is missing

Bug-Debian: https://bugs.debian.org/1002444
Origin: upstream, https://github.com/pypa/setuptools/pull/1986
---
 setuptools/command/install_scripts.py | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/setuptools/command/install_scripts.py b/setuptools/command/install_scripts.py
index 1623427..8c9a15e 100644
--- a/setuptools/command/install_scripts.py
+++ b/setuptools/command/install_scripts.py
@@ -32,8 +32,11 @@ class install_scripts(orig.install_scripts):
         )
         bs_cmd = self.get_finalized_command('build_scripts')
         exec_param = getattr(bs_cmd, 'executable', None)
-        bw_cmd = self.get_finalized_command("bdist_wininst")
-        is_wininst = getattr(bw_cmd, '_is_running', False)
+        try:
+            bw_cmd = self.get_finalized_command("bdist_wininst")
+            is_wininst = getattr(bw_cmd, '_is_running', False)
+        except ImportError:
+            is_wininst = False
         writer = ei.ScriptWriter
         if is_wininst:
             exec_param = "python.exe"
